src/lsm/segmented_array.zig

```diff
@@ -227,8 +227,8 @@ fn SegmentedArrayBaseType(
 
                 return absolute_index;
             }
-        } else struct {
-            pub fn insert_elements(
```

Comment:
Maybe have both `insert_element / insert_elements` and `comptime assert` that the Key is/isn't null?